photon-ml/src/main/scala/com/linkedin/photon/ml/TaskType.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml
+
+import com.linkedin.photon.ml.supervised.classification.{LogisticRegressionModel, SmoothedHingeLossLinearSVMModel}
+import com.linkedin.photon.ml.supervised.model.GeneralizedLinearModel
+import com.linkedin.photon.ml.supervised.regression.{LinearRegressionModel, PoissonRegressionModel}
+
+/**
+ * Type of models supported by Game
+ */
+object TaskType extends Enumeration {
+
+  type TaskType = Value
+  val LINEAR_REGRESSION, POISSON_REGRESSION, LOGISTIC_REGRESSION, SMOOTHED_HINGE_LOSS_LINEAR_SVM, NONE = Value
+
+
+  /**
+   * Decodes the type of a GeneralizedLinearModel to something human readable and independent of our implementation.
+   *
+   * @param model The model for which we want the "type"
+   * @return A human readable string for the model type
+   */
+  def name(model: GeneralizedLinearModel): String =
+    model match {
+      case _: LinearRegressionModel => LINEAR_REGRESSION.toString
+      case _: LogisticRegressionModel => LOGISTIC_REGRESSION.toString
+      case _: PoissonRegressionModel => POISSON_REGRESSION.toString
+      case _: SmoothedHingeLossLinearSVMModel => SMOOTHED_HINGE_LOSS_LINEAR_SVM.toString
+      case _ => "Unknown model type"
+    }
+
+  /**
+   * Checks if a subclass of GeneralizedLinearModel corresponds to the expected TaskType.
+   *
+   * @param model The model to check
+   * @param taskType The expected task type
+   * @return True if the model corresponds to the expected task type, false otherwise
+   */
+  def matches(model: GeneralizedLinearModel, taskType: TaskType): Boolean =
+    (model, taskType) match {
+      case (_: LinearRegressionModel, LINEAR_REGRESSION) => true
+      case (_: LogisticRegressionModel, LOGISTIC_REGRESSION) => true
+      case (_: PoissonRegressionModel, POISSON_REGRESSION) => true
+      case (_: SmoothedHingeLossLinearSVMModel, SMOOTHED_HINGE_LOSS_LINEAR_SVM) => true
+      case (_, _) => false
+    }
```

Comment:
I'm backpedaling on this - I don't want to duplicate what the class hierarchy is already telling us. I'll keep function in TaskType, but I'm reworking them to be lighter.